src/hooks/useCmsLanguageOptions.ts

```diff
@@ -0,0 +1,26 @@
+import { useLocation } from 'react-router';
+
+import { usePageQuery } from './usePageQuery';
+
+// as we dont have /cms-page/ path for cms pages, and we cannot identify,
+// if its a cms page url or not, we use location.pathname as parameter
+// and handle the non cms pages in Navigation component routing utils
+export const useCmsLanguageOptions = ({
+  skip = false,
+}: { skip?: boolean } = {}) => {
+  const location = useLocation();
+  const { data: pageData } = usePageQuery(location.pathname, { skip });
+
+  return !skip
+    ? [
+        {
+          uri: pageData?.page?.uri,
+          locale: pageData?.page?.language?.code?.toLowerCase(),
+        },
+        ...(pageData?.page?.translations?.map((translation) => ({
+          uri: translation?.uri,
+          locale: translation?.language?.code?.toLowerCase(),
+        })) ?? []),
+      ]
+    : [];
+};
```

Comment:
This looks quite bewildering, just the nesting of different kinds of braces, ternary operator, anonymous functions etc. Looks like this code or something similar is already used in palvelutarjotin-ui and palvelutarjotin-admin... but still, looks quite bewildering, confusing. Could be refactored to add readability later.

Reply:
yes, its the same, lets refactor in all projects in other Prs